src/request.rs

```diff
@@ -638,6 +638,27 @@ impl<T> Request<T> {
     pub fn into_parts(self) -> (Parts, T) {
         (self.head, self.body)
     }
+
+    /// Consumes the request returning a new request with body mapped to the
+    /// return type of the passed in function.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # use http::*;
+    /// let request = Request::builder().body("some string").unwrap();
+    /// let mapped_request: Request<&[u8]> = request.map(|b| {
+    ///   assert_eq!(b, "some string");
+    ///   b.as_bytes()
+    /// });
+    /// assert_eq!(mapped_request.body(), &"some string".as_bytes());
+    /// ```
+    #[inline]
+    pub fn map<O, F>(self, op: F) -> Request<O>
```

Comment:
Super nitty, but Rust convention seems to be naming the function argument of a functional combinator `f` (and this will show up in the rustdoc).

Could you rename `op` -> `f`. Also, could you rename `O` -> `U`.

(I am mostly going by observations that I have made of std).

Reply:
Yes of course! I actually had it like that originally but ended up changing it because the `Result` struct in std uses `op`:
https://doc.rust-lang.org/std/result/enum.Result.html#method.map

Reply:
I'm also going to swap `U` and `F` as type arguments `map<F, U>` since the function type appears first in the signature.